### ISSUE TITLE
Is Mesh Gateway necessary for federation to work

### ISSUE THREAD

**Issue by narendrapatel** [hashicorp/consul-k8s#1253]
Hi,
Are Mesh Gateways mandatory to establish federation?

We have direct connectivity available between our VM cluster and K8s cluster and as such want to avoid Mesh Gateways, 
because:
- It adds extra cost in terms of Mesh Gateway nodes(VM) and pods(K8s) and consumes resources
- In K8s, Mesh Gateway is implemented as a Loadbalancer type service which incurs LoadBalancer usage costs
- Data plane traffic is routed through Mesh Gateways which adds latency (we have option to set mesh gateway mode to remote or none)
- Need planning in terms of scaling and monitoring of Mesh Gateway resources

As such due to above factors, we decided to skip federation via Mesh Gateways. Instead we federated with below config:
 ```
  extraConfig: |
    {
      "primary_datacenter": "primary",
      "retry_join_wan":["primary01", "primary02", "primary03"]
    }
```
It worked well, but we ran into issues with upgrading the helm chart from 0.41.1 to 0.44.0, with introduction of auth-methods for controller pods. Exact issue is mentioned in detail over [here](https://discuss.hashicorp.com/t/consul-sidecar-injection-not-working/39974).

Do we have any alternatives to Mesh Gateways or plans to support other ways for federation?

**Comment by david-yu** [comment]
Hi @narendrapatel thanks for filing the issue. Is there a reason why you have VMs and K8s cluster deployed as distinct Consul datacenters? We have seen such a request previously from a customer to improve the overall UX for federation without mesh gateways but this has not been prioritized yet.

**Comment by lkysow** [comment]
Mesh gateways aren't mandatory technically but we use their existence to know if federation is happening. As a workaround you might be able to enable mesh gateways but set their replica count to 0:

```yaml
meshGateways:
  enabled: true
  replicas: 0
```

**Comment by narendrapatel** [comment]
Hi @david-yu ,

For clusters, that's how we started. We already had the VM cluster, so for evaluating service mesh in K8s we installed a Consul cluster there with federation to the VM cluster. This was to avoid any issues from creeping into already existing VM cluster during our POC phase. Moreover having a complete cluster in K8s would have kept the setup gracefully operational even when there is some network disruption between the 2 clusters.

When we encountered the issue mentioned in this ticket, we switched to using the VM cluster as `externalServers` for K8s cluster. This is because our VM and K8s cluster are in the same network and we now have a good understanding of service mesh.

@lkysow thank you for the workaround :). For now, we've solved this with external servers. This helps us remove non required mesh configurations as described in https://github.com/hashicorp/consul-k8s/issues/1022. However this workaround should be helpful for us in federating our DR clusters located in different network locations.

David & Luke, thank you for your time and inputs :) We can consider this as closed.
